### ISSUE TITLE
Entity collections in Hibernate 6 can't be replaced

### ISSUE THREAD

**Issue by ahus1** [keycloak/keycloak#19162]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

storage

### Describe the bug

When planning the Hibernate 6 / Quarkus 3 upgrade, we've found https://hibernate.atlassian.net/browse/HHH-16337

Due to that, for example a client can't be updated due to the `@EntityCollections` 

### Version

quarkus3 branch

### Expected behavior

Should work like in Hibernate 5

### Actual behavior

Fails with duplicate key exceptions as entities as the old entries are not removed:

```
Caused by: org.h2.jdbc.JdbcSQLIntegrityConstraintViolationException: Unique index or primary key violation: "PUBLIC.PRIMARY_KEY_99 ON PUBLIC.REDIRECT_URIS(CLIENT_ID, ""VALUE"") VALUES ( /* key:19 */ '9ecdb6da-5621-47fb-9b0c-99f80564f043', '')"; SQL statement:
insert into REDIRECT_URIS (CLIENT_ID,VALUE) values (?,?) [23505-214]
        at org.h2.message.DbException.getJdbcSQLException(DbException.java:508)
        at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
        at org.h2.message.DbException.get(DbException.java:223)
        at org.h2.message.DbException.get(DbException.java:199)
        at org.h2.index.Index.getDuplicateKeyException(Index.java:525)
        at org.h2.mvstore.db.MVSecondaryIndex.checkUnique(MVSecondaryIndex.java:223)
        at org.h2.mvstore.db.MVSecondaryIndex.add(MVSecondaryIndex.java:184)
        at org.h2.mvstore.db.MVTable.addRow(MVTable.java:519)
        at org.h2.command.dml.Insert.insertRows(Insert.java:174)
        at org.h2.command.dml.Insert.update(Insert.java:135)
        at org.h2.command.dml.DataChangeStatement.update(DataChangeStatement.java:74)
        at org.h2.command.CommandContainer.update(CommandContainer.java:169)
        at org.h2.command.Command.executeUpdate(Command.java:252)
        at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:209)
        at org.h2.jdbc.JdbcPreparedStatement.executeUpdate(JdbcPreparedStatement.java:169)
        at io.agroal.pool.wrapper.PreparedStatementWrapper.executeUpdate(PreparedStatementWrapper.java:88)
        at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:275)
        at org.hibernate.engine.jdbc.mutation.internal.AbstractMutationExecutor.performNonBatchedMutation(AbstractMutationExecutor.java:98)
        at org.hibernate.engine.jdbc.mutation.internal.MutationExecutorSingleNonBatched.performNonBatchedOperations(MutationExecutorSingleNonBatched.java:40)
        at org.hibernate.engine.jdbc.mutation.internal.AbstractMutationExecutor.execute(AbstractMutationExecutor.java:43)
        at org.hibernate.persister.collection.mutation.InsertRowsCoordinatorStandard.insertRows(InsertRowsCoordinatorStandard.java:118)
        at org.hibernate.persister.collection.BasicCollectionPersister.recreate(BasicCollectionPersister.java:112)
        at org.hibernate.action.internal.CollectionRecreateAction.execute(CollectionRecreateAction.java:47)
        at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:616)
      
```

### How to Reproduce?

* Open the Admin UI
* Create a new client and save it
* Open the client again and save it -> Exception

Minimal reproducer for the Hibernate team: https://github.com/ahus1/hibernate-test-case-templates/tree/is-HHH-xxxx-element-collection-after-skipped-autoflush

**Comment by ahus1** [comment]
Fix available upstream, should be part of Hibernate 6.2 release. Already tested the snapshot that it works as expected for us.

**Comment by ahus1** [comment]
The workaround has been removed from the Q3 branch in PR #19486, therefore this is now resolved.
